src/thunderbots/software/radio_communication/visitor/mrf_primitive_visitor.cpp

```diff
@@ -109,7 +109,8 @@ void MRFPrimitiveVisitor::visit(const MoveSpinPrimitive &movespin_primitive)
     radio_prim->param_array = {
         movespin_primitive.getDestination().x() * MILLIMETERS_PER_METER,
         movespin_primitive.getDestination().y() * MILLIMETERS_PER_METER,
-        movespin_primitive.getAngularVelocity().toRadians() * CENTIRADIANS_PER_RADIAN, 0};
+        movespin_primitive.getAngularVelocity().toRadians() * CENTIRADIANS_PER_RADIAN,
+        movespin_primitive.getFinalSpeed() * MILLIMETERS_PER_METER};
```

Comment:
Have we tested this on actual robots with radio?

Reply:
No it hasn't been tested on actual robots. What I thought is that the 0 beforehand is a placeholder for `final_speed` so I just change that. How to test this with actual robots and radio?

Reply:
Ping @henrymliu real quick on slack, he should be able to help you out!

Reply:
tested on field, it works